recording/docker-compose/dockerfile-talk-recording

```diff
@@ -0,0 +1,39 @@
+FROM phusion/baseimage:focal-1.2.0
```

Comment:
I would prefer to use an official Ubuntu image. What benefits does _phusion/baseimage_ provide (in the case of the recording server)? If I am not mistaken the PID 1 zombie reaping problem should be fixed by [launching Tini with `--init`](https://github.com/krallin/tini/#using-tini).

Reply:
The official docs have a comprehensive list of differences: https://github.com/phusion/baseimage-docker#overview

Personally these are the reasons why I didn't go with a bare-bones Ubuntu image:
- Simplified management of services. Auto-restart, running as another user.
- Built-in logger that plays nice with docker.
- Possibility to run multiple services at once.
- A working CRON daemon.

Anyway, those are _personal_ preferences in the end, and I understand if an untouched Ubuntu image is preferred.

Reply:
From my point of view, while those can be useful features, in the case of a container for the recording server they are not strictly needed:
- The recording server should always run as its own user, which (if I recall correctly) can be set in the Dockerfile and overriden if needed when creating the container with `docker run --user XXX`. Auto-restarting can also be set when creating the container with `docker run --restart on-failure`.
- As the recording server is directly run and it prints to stdout the logs should be easily redirected to a proper logging system on the host, like [_journald_](https://docs.docker.com/config/containers/logging/journald/)
- Only a single service, the recording server, is expected to be run inside the container; other services (like a web server to provide TLS termination for the recording server) should be run in other containers or the host
- CRON is not used by the recording server, so it should not be needed in its container

Of course all this is based also on my personal view on how the container for the recording server should work, but for maintenance reasons I would prefer something as simple and generic as possible :-)

Reply:
I'm managing my docker containers with `docker compose`, not with `docker run`.

My reason for this is to be able to build everything with `docker compose build`, and then run a simple `docker compose up` for starting everything (and `docker compose down` for the opposite).

I get the feeling the suggestions go towards a `docker run` approach, with a single `Dockerfile`, and no `docker-compose.yml`.

The `docker compose` use-case is kinda specific to my environment, and I'm unsure how useful would it be to others (and have it included as part of the spreed/Talk source code).

In the end, I think the question can be resumed to a logical or:
- Improve [start-container.sh](https://github.com/nextcloud/spreed/blob/5c5588a460c115c402ce025222b0baa67c964835/recording/start-container.sh)? (personally, I see it in a good enough state).
- Add a separate `docker compose` implementation (this PR)?
- Something else?

No hard feelings if the `docker compose` approach is undesirable (:

Reply:
I used `docker run` as an example, but [`user`](https://docs.docker.com/compose/compose-file/05-services/#user) and [`restart`](https://docs.docker.com/compose/compose-file/05-services/#restart) can be used too in Docker compose (and probably anything that can be passed to `docker run`). And even if I start my containers with `docker run` for development purposes I would like to have a Docker compose file :-) After all the Docker compose file can be seen as an implicit documentation of the parameters that need to be given to `docker run` ;-)

Actually I always wanted to provide a Dockerfile in addition to `start-container.sh`; `start-container.sh` was modified from a hacky script I was using when the recording server development started, but although it works for development I think that being able to build a proper image would be better for end "users".